nixos/modules/services/mail/dovecot.nix

```
@@ -692,23 +692,69 @@ in
 
     environment.etc."dovecot/dovecot.conf".source = cfg.configFile;
 
-    systemd.services.dovecot2 = {
+    systemd.services.dovecot = {
```

Renaming the systemd service also broke my config because i had overrides setup for dovecot2.service.
We should at least include an assertion or something on the old service name to make it fail for people still using it.

---

Added an assertion when `config.systemd.services.dovecot2` is set.

---

Just out of curiosity, why even rename it at all?

---

This is the reason why I originally suggested 


```suggestion
    systemd.services.dovecot2 = {
      name = "dovecot";
```

btw; however ideally of course the attr and service name match up.

> Now I Just need to find a clean way in nixos-mailserver to understand whether dovecot or dovecot2 is the unit name used.

Add an internal option services.dovecot2.serviceNameIsDovecot, set it to true here and check whether it exists and is true in nixos-mailserver? Kinda awful but works (:

---

Added in 4424fa674fc5